crates/dojo-core/src/world.cairo

```diff
@@ -35,23 +43,108 @@ mod World {
         executor: ContractAddress,
         component_registry: LegacyMap::<ShortString, ClassHash>,
         system_registry: LegacyMap::<ShortString, ClassHash>,
+        initialized: bool,
         nonce: usize,
     }
 
-    // Give deployer the default admin role.
     #[constructor]
     fn constructor(name: ShortString, executor_: ContractAddress) {
         executor::write(executor_);
 
-        WorldSpawned(get_contract_address(), name);
+        WorldSpawned(get_contract_address(), get_tx_info().unbox().account_contract_address, name);
+    }
+
+    // Initialize the world with the routes that specify
+    // the permissions for each system to access components.
+    #[external]
+    fn initialize(routes: Array<Route>) {
+        // Assert that the world has not been initialized
+        assert(!initialized::read(), 'already initialized');
+
+        let route_auth_class_hash = system_registry::read('RouteAuth'.into());
+        let mut index = 0;
+
+        // Loop through each route and handle the auth.
+        // This grants the system the permission to specific components.
+        loop {
+            if index == routes.len() {
+                break ();
+            }
+
+            // Serialize the route
+            let mut calldata = ArrayTrait::new();
+            let r = routes.at(index);
+            r.serialize(ref calldata);
+
+            // Call RouteAuth system via executor with the serialized route
+            IExecutorDispatcher {
+                contract_address: executor::read()
+            }.execute(route_auth_class_hash, calldata.span());
+
+            index += 1;
+        };
```

Comment:
Looping through the array of routes and calling the executor to call the `RouteAuth` system to set the authentication routes of the world